.github/workflows/ci-publish.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish to NPM
+on: 
+  push:
+    tags:
+      - '*'
+
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: install
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: test
+
+  publish-npm:
+    needs: build
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: build
+      - uses: borales/actions-yarn@v2.0.0
```

Comment:
Would like to only execute this step if it is a tag; that way only one workflow definition is needed and we can expand the `on` to be both branches and tags. Checking what kind of `if` is needed here.